src/visual.ts

```diff
@@ -573,6 +556,35 @@ export class BulletChart implements IVisual {
         return barData;
     }
 
+    private static computeCategoryNumbers(categoricalValues: BulletChartColumns<any[]>, idx: number, visualSettings: BulletChartSettingsModel, targetValue: number, minimum: number, categoryMaxValue: number, categoryValue: number, targetValue2: number) {
+        let needsImprovement: number = BulletChart.GETRANGEVALUE(categoricalValues.NeedsImprovement ? categoricalValues.NeedsImprovement[idx] : undefined, visualSettings.values.needsImprovementPercent.value, targetValue, minimum);
+        let satisfactory: number = BulletChart.GETRANGEVALUE(categoricalValues.Satisfactory ? categoricalValues.Satisfactory[idx] : undefined, visualSettings.values.satisfactoryPercent.value, targetValue, minimum);
```

Comment:
refactor function GET_RANGE_VALUE

Reply:
categoricalValues.Satisfactory ? categoricalValues.Satisfactory[idx] : undefined check this inside function

Reply:
cover with tests